tests/testthat/test-brms_compare.R

```diff
@@ -0,0 +1,251 @@
+
+
+test_that("jmpost and brms get similar loo for longitudinal models", {
+
+    skip_if_not(is_full_test())
+
+    set.seed(22231)
+    n <- 250
+    mu_b <- 60
+    mu_s <- 0.5
+    mu_g <- 0.2
+    omega_b <- 0.1
+    omega_s <- 0.1
+    omega_g <- 0.1
+    sigma <- 1.5
+    n_vis <- 4
+
+    sf_sld <- function(time, b, s, g) {
+        s <- dplyr::if_else(time >= 0, s, 0)
+        b * (exp(-s * time) + exp(g * time) - 1)
+    }
+
+
+    baseline <- dplyr::tibble(
+        pt = sprintf("pt_%06i", seq_len(n)),
+        b = rlnorm(n, log(mu_b), omega_b),
+        s = rlnorm(n, log(mu_s), omega_s),
+        g = rlnorm(n, log(mu_g), omega_g),
+    )
+
+    dat_full <- dplyr::tibble(
+        pt = rep(sprintf("pt_%06i", seq_len(n)), each = n_vis),
+        time = rep(seq(0, 3, length.out = n_vis), n)
+    ) |>
+        dplyr::left_join(baseline, by = "pt") |>
+        dplyr::mutate(mu = sf_sld(time = time, b = b, s = s, g = g)) |>
+        dplyr::mutate(value = rnorm(n * n_vis, mu, sigma))
+
+    dat <- dat_full |>
+        dplyr::select(pt, value, time)
+
+
+    # nolint start
+    # DEBUG
+    # ggplot(data = filter(dat, pt %in% sample(dat$pt, 5)), aes(x = time, y = value, col = pt, group = pt)) +
+    #     geom_point() +
+    #     geom_line()
+    # nolint end
+
+    dat2 <- dat |>
+        dplyr::mutate(arm = "A") |>
+        dplyr::mutate(study = "A")
+
+    dat_bl <- dat2 |>
+        dplyr::select(pt, arm, study) |>
+        dplyr::group_by(pt) |>
+        dplyr::slice(1) |>
+        dplyr::ungroup()
+
+
+    jdat <- DataJoint(
+        subject = DataSubject(
+            data = dat_bl,
+            subject = "pt",
+            arm = "arm",
+            study = "study"
+        ),
+        longitudinal = DataLongitudinal(
+            data = dat2,
+            formula =  value ~ time,
+            threshold = -99999
+        )
+    )
+
+
+    jm <- JointModel(
+        longitudinal = LongitudinalSteinFojo(
+            mu_bsld = prior_normal(log(mu_b), 0.6),
+            mu_ks = prior_normal(log(mu_s), 0.6),
+            mu_kg = prior_normal(log(mu_g), 0.6),
+            omega_bsld = prior_lognormal(log(omega_b), 0.6),
+            omega_ks = prior_lognormal(log(omega_s), 0.6),
+            omega_kg = prior_lognormal(log(omega_g), 0.6),
+            sigma = prior_lognormal(log(sigma), 0.6),
+            centred = FALSE,
+            scaled_variance = FALSE
+        )
+    )
+
+    mp <- sampleStanModel(
+        jm,
+        data = jdat,
+        iter_warmup = 1400,
+        iter_sampling = 2600,
+        chains = 2,
+        refresh = 200,
+        parallel_chains = 2
+    )
+
+
+    stanmod <- as.CmdStanMCMC(mp)
+
+
+    mp_brms <- brms::brm(
```

Comment:
I guess strictly speaking these tests should have a `skip_if_not(requireNamespace('brms')`
@gowerc

Reply:
As discussed, leave as is